hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/GenesisRecordsConsensusHook.java

```diff
@@ -160,6 +167,19 @@ private void createAccountRecordBuilders(
                     Transaction.newBuilder().body(TransactionBody.newBuilder().cryptoCreateAccount(txnBody));
             recordBuilder.transaction(txnBuilder.build());
 
+            var balance = account.tinybarBalance();
+            if (balance != 0) {
+                var accountID = AccountID.newBuilder()
+                        .accountNum(account.accountId().accountNumOrElse(0L))
+                        .shardNum(account.accountId().shardNum())
+                        .realmNum(account.accountId().realmNum())
+                        .build();
+
+                recordBuilder.transferList(TransferList.newBuilder()
+                        .accountAmounts(asAccountAmounts(Map.of(accountID, balance)))
+                        .build());
```

Comment:
Is this for funding the treasury? (If not, what is it for?)

Reply:
@mhess-swl it was added, so`transferList` in the item below from the record stream generated by running a HapiTest network corresponds to the same item from the record stream generated by a standalone mono-service node (hence the balance validation passes).

```
record_stream_items {
  transaction {
    signedTransactionBytes: "\nN\n\021\n\t\b\250\353\231\255\006\020\360\003\022\002\030\002 \002Z9\n\"\022 \n\250\342\020d\306\036\253\206\342\251\301dV[Nz\232AF\020n\nl\320:\2149Z\021\016\222\020\200\200\320\247\244\260\344\261EJ\006\b\366\317\252\340\006\210\001\001"
  }
  record {
    receipt {
      accountID {
        accountNum: 2
      }
    }
    transactionHash: "8\260`\247Q\254\2268L\3312~\261\261\343j!\375\267\021\024\276\aCL\f\307\277c\366\341\332\'N\336\277\347oe\373\325\032\322\361H\230\271["
    consensusTimestamp {
      seconds: 1705407972
      nanos: 258727001
    }
    transactionID {
      transactionValidStart {
        seconds: 1705407912
        nanos: 496
      }
      accountID {
        accountNum: 2
      }
      nonce: 2
    }
    memo: "Synthetic system creation"
    transferList {
      accountAmounts {
        accountID {
          accountNum: 2
        }
        amount: 5000000000000000000
      }
    }
  }
}
```

Reply:
Ahh yes, @Neeharika-Sompalli should we build this record ☝️  somewhere else? It seemed appropriate to build in the consensus hook to me